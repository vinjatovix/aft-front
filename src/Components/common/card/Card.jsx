import React from "react";
import PropTypes from "prop-types";
import { UserCard } from "./UserCard";

const Card = ({ data, type, actions, isAdmin, token }) => {
  return (
    <div className="card" onClick={() => actions.setDataDetail(data)}>
      {type === "user" && <UserCard data={data} />}
    </div>
  );
};

Card.propTypes = {
  type: PropTypes.string.isRequired,
  data: PropTypes.object.isRequired,
  actions: PropTypes.object,
  isAdmin: PropTypes.bool,
  token: PropTypes.string,
};

Card.defaultProps = {
  isAdmin: false,
};

export default Card;
