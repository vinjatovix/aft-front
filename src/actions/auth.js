import { ACTION_TYPES } from "../types/types";

export const loginRequestAction = () => ({
  type: ACTION_TYPES.AUTH_LOGIN_REQUEST,
});

export const loginSuccessAction = (auth) => ({
  type: ACTION_TYPES.AUTH_LOGIN_SUCCESS,
  payload: auth,
});

export const loginFailureAction = (error) => ({
  type: ACTION_TYPES.AUTH_LOGIN_FAILURE,
  payload: error,
});

export const logoutAction = () => ({
  type: ACTION_TYPES.AUTH_LOGOUT,
});
