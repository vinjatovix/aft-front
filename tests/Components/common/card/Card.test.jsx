import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Card from "../../../../src/Components/common/card/Card";
import actions from "../../../fixtures/actions";

describe("Card Component", () => {
  it("should render a book card", () => {
    const { container } = render(
      <>
        <Router>
          <Card
            type="book"
            data={{
              _id: "5e9f8f8f8f8f8f8f8f8f8f8",
              name: "Obra",
              author: "Autor",
              img: "https://boxshot.com/3d-book-cover/how-to-make-a-3d-book-cover-in-photoshop/sample.jpg",
              description: "Descripción",
            }}
            actions={actions}
            isAdmin={false}
            token="token"
          />
        </Router>
      </>
    );

    expect(container).toMatchSnapshot();
  });

  it("should render a works card", () => {
    const { container } = render(
      <>
        <Router>
          <Card
            type="works"
            data={{
              _id: "5e9f8f8f8f8f8f8f8f8f8f8",
              book: { _id: "5e9f8f8f8f8f8f8f8f8f8f8", name: "Obra" },
              student: { _id: "5e9f8f8f8f8f8f8f8f8f8f8", name: "Alumno" },
              character: { _id: "5e9f8f8f8f8f8f8f8f8f8f8", name: "Personaje" },
              scene: { _id: "5e9f8f8f8f8f8f8f8f8f8f8", name: "Escena" },
              description: "Descripción",
            }}
            actions={actions}
            isAdmin={false}
            token="token"
          />
        </Router>
      </>
    );

    expect(container).toMatchSnapshot();
  });
});
