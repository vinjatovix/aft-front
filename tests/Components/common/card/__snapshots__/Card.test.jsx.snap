// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should render a book card 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="flex-wrapper column"
    >
      <ul
        class="book"
        data-testid="book-detail"
      >
        <img
          alt="Obra"
          src="https://boxshot.com/3d-book-cover/how-to-make-a-3d-book-cover-in-photoshop/sample.jpg"
        />
        <li
          class=""
          data-testid="Obra"
        >
          Obra
        </li>
        <li
          class="small"
          data-testid="Autor"
        >
          Autor
        </li>
      </ul>
      <a
        class=""
        href="/characters/book/5e9f8f8f8f8f8f8f8f8f8f8"
      >
        <button
          class="bg-blue"
          data-testid="navigate-button"
        >
          ðŸŽ­ Personajes
        </button>
      </a>
      <a
        class=""
        href="/scenes/book/5e9f8f8f8f8f8f8f8f8f8f8"
      >
        <button
          class="bg-blue"
          data-testid="navigate-button"
        >
          ðŸŽ¬ Escenas
        </button>
      </a>
    </div>
  </div>
</div>
`;

exports[`Card Component should render a works card 1`] = `
<div>
  <div
    class="card"
  />
</div>
`;
